public/app/features/explore/LogsContainer.tsx

```diff
@@ -87,6 +87,11 @@ export class LogsContainer extends PureComponent<LogsContainerProps> {
     return [];
   };
 
+  deduplicateLogRows = (rows: LogRowModel[]) => {
+    // Deduplicate log rows with the same log uid
+    return rows.filter((row, idx, self) => idx === self.findIndex(r => r.uid === row.uid));
```

Comment:
I would probably just use lodash uniq here. Seems like it has some optimisations depending on array size and will probably scale a bit better.